packages/ui/src/forum/components/Thread/WatchlistButton.tsx

```diff
@@ -21,7 +21,7 @@ export const WatchlistButton = ({ threadId }: Props) => {
   const removeFromWatchlist = useCallback(
     (e) => {
       e.preventDefault()
-      setWatchlist(watchlist?.filter((id) => id !== threadId) ?? [])
+      if (watchlist) setWatchlist([...watchlist, threadId].filter((id) => id !== threadId))
```

Comment:
The idea behind this issue is that by the time `addToWatchlist` or `removeFromWatchlist` are called, the `watchlist` variable might be out of date due to changes made from other window (or potentially other components since `useLocalStorage` doesn't rely on any reactive global context AFAIK although it's probably irrelevant for this use case).

So the proposed solution is to pass an update function to `setWatchlist` (similarly to how react `useState` works [here](https://beta.reactjs.org/apis/react/useState#updating-state-based-on-the-previous-state)). The key is that this update function is given the value of the local storage at the time of the update opposed to the time of the component render (incidentally it also updates the value for the rest of the component after being called).

This way this line would be:
```suggestion
      setWatchlist((watchlist = []) => watchlist.filter((id) => id !== threadId))
```

And line 17:
```ts
setWatchlist((watchlist = []) => [...watchlist, threadId])
```